scripts/release/index.ts

```diff
@@ -0,0 +1,150 @@
+import fs from 'fs';
+import { getPackageType, getCustomPackagesInIntegrationWrapper } from './packageTypes';
+import updatePackageVersion from './updatePackageVersion';
+import { execSync } from 'child_process';
+import { RELEASE_GRADATIONS, PACKAGE_SUBTYPE, PACKAGE_TYPES, REGISTRY } from './constants';
+
+const publishPackage = (path: string) => execSync(`cd ${path} && npm publish --registry ${REGISTRY}`);
+const isProperGradation = (gradation: string) => Object.keys(RELEASE_GRADATIONS).filter(key => isNaN(Number(key))).includes(gradation);
+const sliceAbsolutePathPart = (path: string) => path.replace(/(.*?)packages/, 'packages');
+
+// CLI          depends on THEME
+// THEME        depends on COMPOSABLE
+// COMPOSABLE   depends on API
+
+const getPackageSubtype = (pckg: string): PACKAGE_SUBTYPE => {
+  switch (pckg.split('/').pop()) {
+    case 'api-client':
+      return PACKAGE_SUBTYPE.API;
+    case 'composables':
+      return PACKAGE_SUBTYPE.COMPOSABLE;
+    case 'theme':
+      return PACKAGE_SUBTYPE.THEME;
+    case 'cli':
+      return PACKAGE_SUBTYPE.CLI;
+    default:
+      return PACKAGE_SUBTYPE.INDEPENDENT;
+  }
+};
+
+const getSubtypeName = (subtype: PACKAGE_SUBTYPE): string => {
+  switch (subtype) {
+    case PACKAGE_SUBTYPE.API:
+      return 'api-client';
+    case PACKAGE_SUBTYPE.COMPOSABLE:
+      return 'composables';
+    case PACKAGE_SUBTYPE.THEME:
+      return 'theme';
+    case PACKAGE_SUBTYPE.CLI:
+      return 'cli';
+    default:
+
+  }
+};
+
+const pckgSubtypeToPath = (pckg: string, subtype: PACKAGE_SUBTYPE): string => {
+  if (subtype === PACKAGE_SUBTYPE.CLI) {
+    return 'core/cli';
+  }
+  const base = pckg.split('/');
+  base[base.length - 1] = getSubtypeName(subtype);
+  return base.join('/');
+};
+
+const pckgCustomSubpackageToPath = (pckg: string, subpackage: string): string => {
+  const base = pckg.split('/');
+  base[base.length - 1] = subpackage;
+  return base.join('/');
+};
+
+const buildPackageDependencyList = (pckg: string): PACKAGE_SUBTYPE[] => {
```

Comment:
```
const buildPackageDependencyList = (pckg: string): PACKAGE_SUBTYPE[] => {
  const packageSubtype = getPackageSubtype(pckg);
  if (packageSubtype === PACKAGE_SUBTYPE.INDEPENDENT) {
    return [];
  }

  const enumKeys = Object.keys(PACKAGE_SUBTYPE)
    .filter(key => isNaN(Number(key)))

  return Array.from(enumKeys.keys());
};
```

Reply:
but please check

Reply:
It won't work, look at this part:
```
for (let i = packageSubtype + 1; i < enumKeys.length; i++) {
```
It does not use whole array. It begins from `packageSubtype + 1`. I want to create array of values from X to Y. Like: [X, X+1, X+2, Y-1. Y] (in example Y = X+4). Your example would create array from 0 to Y but always X > 0.

Reply:
> I want to create array of values from X to Y.

```
Array.from([1, 2, 3, 4, 5].keys()).slice(X)
```